### PR TITLE
RDK-33287: Stop Maintenance API

### PR DESCRIPTION
**Pull request by livinsunny519**
Reason for change: Abort the maintenance task if an
api call to thunder plugin.
Test Procedure: Build and verify
Run all end to end scenarios.
Risk: High

### CODE REVIEW

**Comment by mhughesacn** [comment]
Hi @livinsunny519
MaintenanceManager/MaintenanceManager.cpp:findProcessPid is not original code.  I think the original source is http://proswdev.blogspot.com/2012/02/get-process-id-by-name-in-linux-using-c.html and this is a problem as there is no permission to use (open to correction).  Comcast Legal has prevented such code from being used before.  The code is a straight copy and must be rewritten.  However, this is a pretty standard way to read the proc file, so it is acknowledged that rewritten code will have to do similar actions.  But comments and variable names must not be identical, or layout as much as possible.
Thanks,
Martin Hughes RDK CMF Compliance Team

**Comment by livinsunny519** [comment]
This is just for checking the code logic.. This will be removed.

**Comment by mhughesacn** [comment]
Thanks Sunny

File: MaintenanceManager/MaintenanceManager.cpp [hunk 1]
```diff
@@ -874,6 +888,163 @@ namespace WPEFramework {
...
+            int k_ret=EINVAL;
+            int i=0,record=-1;
+            int32_t exec_status=E_NOK;
+
+            bool task_status[4]={false};
+            bool result=false;
+            bool task_incomplete=false;
+
+            /* only unsolicited maintenance we honour this call & based on RFC */
+            if( checkAbortFlag() && (UNSOLICITED_MAINTENANCE == g_maintenance_type )){
```
Comment (livinsunny519) [inline MaintenanceManager/MaintenanceManager.cpp:913]: need to be removed. Only RFC check is fine.

File: MaintenanceManager/MaintenanceManager.cpp [hunk 2]
```diff
@@ -874,6 +888,163 @@ namespace WPEFramework {
...
+                            record = i;
+                            LOGINFO("Checking the PID\n");
+                            pid_num=getTaskPID(script_names[i].c_str());
+                            LOGINFO("PID of script_name [%d] = %s is %d \n", i,script_names[i].c_str(),pid_num);
+                            if( pid_num != -1){
+                                /* send the signal to task to terminate */
+                                k_ret=kill(pid_num,SIGTERM);
+                                if (k_ret == 0){
+                                    LOGINFO(" %s Termimated\n",script_names[i].c_str());
+                                    /*this means we killed the task currently running */
```
Comment (livinsunny519) [inline MaintenanceManager/MaintenanceManager.cpp:945]: break the loop if we are done with the PID. No need to continue.

**Review by livinsunny519 — COMMENTED** [review]
addressed

File: MaintenanceManager/MaintenanceManager.cpp [hunk 3]
```diff
@@ -874,6 +885,166 @@ namespace WPEFramework {
...
+                        LOGINFO("task status [%d]  = %s ScriptName %s",i,(task_status[i])? "true":"false",script_names[i].c_str());
+                    for (i=0;i<4;i++){
+                        if(task_status[i]){
+                            record = i;
+                            LOGINFO("Checking the Task PID\n");
+                            pid_num=getTaskPID(script_names[i].c_str());
+                            LOGINFO("PID of script_name [%d] = %s is %d \n", i,script_names[i].c_str(),pid_num);
+                            if( pid_num != -1){
+                                /* send the signal to task to terminate */
+                                k_ret=kill(pid_num,SIGABRT);
```
Comment (anand-ky) [inline MaintenanceManager/MaintenanceManager.cpp:939]: Should this be a SIGABRT or SIGTERM? Is this getting trapped by the scripts for graceful exit?

**Review by anand-ky — CHANGES_REQUESTED** [review]

File: MaintenanceManager/MaintenanceManager.cpp (hunk 3 above)
Comment (livinsunny519) [reply MaintenanceManager/MaintenanceManager.cpp:939]: SIGTERM is getting ignored in the case SOLICITED maintenance
Due to this issue, we moved to SIGABRT
all scripts have signal handlers and exiting gracefully.

Comment (anand-ky) [reply MaintenanceManager/MaintenanceManager.cpp:939]: SIGABRT handled as part of https://gerrit.teamccp.com/#/q/topic:RDK-33288_2105_sprint+(status:open+OR+status:merged)

**Review by anand-ky — APPROVED** [review]

**Pull request merged**